### PR TITLE
[Test-Proxy] Windows Image, Additional Documentation, Certificate Update.

### PR DESCRIPTION
**Pull request by scbedd**
- Lots of documentation updates (around SSL and the like)
- Adding `windows` version of the container, so that our windows machines in devops can load
- Renaming from `ubuntu_testproxy` to `testproxy` because the manifest will figure out which one you need
- Updating the default `pfx` that we install on the docker images to have a password, such that you don't hit issues when you try to import it with `-p ""`. On pshell it can take that as "not passing" a required argument. Lets just avoid that.

### CODE REVIEW

**Comment by scbedd** [comment]
[Confirmed Docker Publishing Additions](https://dev.azure.com/azure-sdk/internal/_build/results?buildId=1034940&view=results)

File: tools/test-proxy/docker/dev_certificate/apply-dev-cert.sh [hunk 1]
```diff
@@ -28,4 +28,4 @@ fi
 $SUDO cp $CRTFILE "/usr/local/share/ca-certificates"
 $SUDO update-ca-certificates
 
-dotnet dev-certs https --clean --import $PFXFILE -p ""
+dotnet dev-certs https --clean --import $PFXFILE -p "password"
```
Comment (weshaggard) [inline tools/test-proxy/docker/dev_certificate/apply-dev-cert.sh:31]: Is that a real password?

File: tools/test-proxy/documentation/methodology.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+# Test-Proxy Introduction
+
+TODO: create 10k foot intro and detail the process for beginning
```
Comment (weshaggard) [inline tools/test-proxy/documentation/methodology.md:3]: I assume you plan to fill in the various TODO's later?

File: tools/test-proxy/startup-scripts/start-server.py [hunk 3]
```diff
@@ -5,7 +5,8 @@
 import os
 
 CONTAINER_NAME = "ambitious_azsdk_test_proxy"
-IMAGE_SOURCE = "azsdkengsys.azurecr.io/engsys/ubuntu_testproxy_server:952205"
+# this image should be pinned when merged to main in your repo.
+IMAGE_SOURCE = "azsdkengsys.azurecr.io/engsys/testproxy:952205"
```
Comment (weshaggard) [inline tools/test-proxy/startup-scripts/start-server.py:9]: Doe these not need to have the "-lin" or "-win" postfix? Where does that selection happen?

File: tools/test-proxy/docker/dev_certificate/apply-dev-cert.sh (hunk 1 above)
Comment (scbedd) [reply tools/test-proxy/docker/dev_certificate/apply-dev-cert.sh:31]: This is a test certificate. Yes.

File: tools/test-proxy/documentation/methodology.md (hunk 2 above)
Comment (scbedd) [reply tools/test-proxy/documentation/methodology.md:3]: Yep.

File: tools/test-proxy/startup-scripts/start-server.py (hunk 3 above)
Comment (scbedd) [reply tools/test-proxy/startup-scripts/start-server.py:9]: Yeah. I just hit discovered an entirely new rabbit hole I need to go down to publish multiple platforms under a single tag. I'll update this to reflect appropriately.

File: tools/test-proxy/docker/dev_certificate/apply-dev-cert.sh (hunk 1 above)
Comment (scbedd) [reply tools/test-proxy/docker/dev_certificate/apply-dev-cert.sh:31]: To be entirely clear, this is _specifically_  an ASP.NET Development certificate, generated for common name `localhost`. It's not really an abusable resource, as `localhost` certificates are strictly not really secure. No browser will ever allow spoofing using this certificate.

File: tools/test-proxy/documentation/methodology.md (hunk 2 above)
Comment (scbedd) [reply tools/test-proxy/documentation/methodology.md]: I just ripped the file out. No reason to pollute the repo until it's done.

**Pull request merged**